### PR TITLE
feat: enable THM coupling with plastic models

### PR DESCRIPTION
**Pull request by jhuang2601**
This PR 
- [x] fix function interface to enable `PorousDruckerPrager` and `PorousExtendedDruckerPrager` and `PorousModifiedCamClay` for thermal coupling
- [x] add smoke tests for these poro-thermo-plastic models to the integratedTests
- [x] enable THM coupling for all available mechanical models

### CODE REVIEW

**Review by herve-gross — APPROVED** [review]
This seems fine, as long as it does not create problems with the compiler. If it passes CI, then it is all good.

**Comment by CusiniM** [comment]
Is everything in our formulation for THM valid for plastic models? I can't think of anything that would be an issue but it's good to think about it.

**Comment by sytuannguyen** [comment]
The xml files seem to share most of the blocks except the ones for the solid constitutive models. It might be profitable to create a base file for all the shared block.

**Review by sytuannguyen — APPROVED** [review]

File: inputFiles/thermoPoromechanics/ThermoPoroDruckerPrager_consolidation_smoke.xml [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+      defaultDensity="2700"
+      defaultBulkModulus="0.5e9"
+      defaultShearModulus="0.3e9"
+      defaultCohesion="0.1e6"
+      defaultFrictionAngle="15.27"
+      defaultDilationAngle="15.0"
+      defaultHardeningRate="10.0e6"
+      defaultDrainedLinearTEC="3e-7"/> 
+
+    <!-- SPHINX_POROSITY -->
```
Comment (paveltomin) [inline inputFiles/thermoPoromechanics/ThermoPoroDruckerPrager_consolidation_smoke.xml]: do you use those `SPHINX` tags somewhere?

**Review by paveltomin — APPROVED** [review]

File: inputFiles/thermoPoromechanics/ThermoPoroDruckerPrager_consolidation_smoke.xml (hunk 1 above)
Comment (jhuang2601) [reply inputFiles/thermoPoromechanics/ThermoPoroDruckerPrager_consolidation_smoke.xml]: No, this was copied from existing tutorial example and should be removed

Comment (paveltomin) [reply inputFiles/thermoPoromechanics/ThermoPoroDruckerPrager_consolidation_smoke.xml]: i would remove them maybe

Comment (jhuang2601) [reply inputFiles/thermoPoromechanics/ThermoPoroDruckerPrager_consolidation_smoke.xml]: Yes, just removed them

**Comment by jhuang2601** [comment]
Expected failure for adding new tests and rebaseline is needed
```
=======================
Test failures
=======================
ThermoPoroDruckerPrager_consolidation_smoke_01
ThermoPoroDruckerPrager_consolidation_smoke_02
ThermoPoroExtendedDruckerPrager_consolidation_smoke_01
ThermoPoroExtendedDruckerPrager_consolidation_smoke_02
ThermoPoroMCC_consolidation_smoke_01
ThermoPoroMCC_consolidation_smoke_02
```

**Comment by jhuang2601** [comment]
@CusiniM @rrsettgast @cssherman Could you please review this PR?

**Review by npillardou — APPROVED** [review]

File: src/coreComponents/physicsSolvers/multiphysics/poromechanicsKernels/PoromechanicsKernels.cmake [hunk 2]
```diff
@@ -113,7 +113,20 @@ endif( )
 
 set( kernelNames ThermoPoromechanicsKernels )
 set( subregionList CellElementSubRegion )
-set( porousSolidDispatch PorousSolid<ElasticIsotropic> )
+set( porousSolidDispatch PorousSolid<DruckerPragerExtended>
```
Comment (CusiniM) [inline src/coreComponents/physicsSolvers/multiphysics/poromechanicsKernels/PoromechanicsKernels.cmake]: if we dispatch on all of them you can just add ThermoPoromechanicsKernels to the first list.

Comment (CusiniM) [reply src/coreComponents/physicsSolvers/multiphysics/poromechanicsKernels/PoromechanicsKernels.cmake]: Like in 
````
set( kernelNames PoromechanicsKernels ThermoPoromechanicsKernels )

Comment (jhuang2601) [reply src/coreComponents/physicsSolvers/multiphysics/poromechanicsKernels/PoromechanicsKernels.cmake]: You mean also removing `set( porousSolidDispatch PorousSolid<DruckerPragerExtended>` and following porousSolidDispatch? But, locally, compilation error was reported for all plasticity models:

/usr/bin/ld: lib/libphysicsSolvers.so: undefined reference to `double geos::thermalPoromechanicsKernels::ThermalMultiphasePoromechanics<geos::CellElementSubRegion, geos::constitutive::PorousSolid<geos::constitutive::ModifiedCamClay>, geos::finiteElement::ConformingVirtualElementOrder1<14, 7> >::kernelLaunch<RAJA::PolicyBaseT<(RAJA::Policy)3, (RAJA::Pattern)1, (RAJA::Launch)0, (camp::resources::v1::Platform)1, RAJA::policy::omp::Parallel, RAJA::wrapper<RAJA::policy::omp::omp_for_schedule_exec<RAJA::policy::omp::Auto> > >, geos::thermalPoromechanicsKernels::ThermalMultiphasePoromechanics<geos::CellElementSubRegion, geos::constitutive::PorousSolid<geos::constitutive::ModifiedCamClay>, geos::finiteElement::ConformingVirtualElementOrder1<14, 7> > >(int, geos::thermalPoromechanicsKernels::ThermalMultiphasePoromechanics<geos::CellElementSubRegion, geos::constitutive::PorousSolid<geos::constitutive::ModifiedCamClay>, geos::finiteElement::ConformingVirtualElementOrder1<14, 7> > const&)'

Comment (CusiniM) [reply src/coreComponents/physicsSolvers/multiphysics/poromechanicsKernels/PoromechanicsKernels.cmake]: I should have fixed it.

**Review by CusiniM — APPROVED** [review]

**Pull request merged**